src/akar/util.clj

```diff
@@ -0,0 +1,42 @@
+(ns akar.util
+  (:require [clojure.walk :as cw]))
+
+(defn postwalk-fn-replace
+  "Calls the matching-fn on every expression and if matching-fn
+  returned true, replaces the expression by the result of calling
+  replacing-fn with the given expr as the parameter"
+  [matching-fn replacing-fn form]
+  (cw/postwalk (fn [x]
+                   (if (matching-fn x)
+                     (replacing-fn x)
+                     x))
+                 form))
+
+
+(defmacro def-trampolined-recur
```

Comment:
Let's call the macro `defn-trampolined`.

Reply:
done